non_profit/non_profit/doctype/member/member.py

```diff
@@ -11,9 +11,57 @@
 from payments.utils import get_payment_gateway_controller
 
 from non_profit.non_profit.doctype.membership_type.membership_type import get_membership_type
+from erpnext.accounts.doctype.subscription.subscription import Subscription
+from erpnext.accounts.doctype.subscription.subscription import is_prorate, get_prorata_factor, process_all
 
 
 class Member(Document):
+    @frappe.whitelist()
+    def create_membership_and_link(self):
+        if self.member_approval_status != "Approved":
+            frappe.msgprint(_("Member is not approved yet"))
+            return
+
+        existing_membership = frappe.get_all(
+            "Membership", filters={"member": self.name})
+        if existing_membership:
+            frappe.msgprint(_("Membership already exists for this Member"))
+            return
+
+        membership = create_membership(frappe._dict({
+            'member': self.name,
+            'membership_type': self.membership_type,
+            'from_date': self.initial_membership,
+            'to_date': self.membership_expiry_date
```

Comment:
how are we getting initial_membership and membership_expiry_date?

This should be fetched from a current  subscription period that todays date falls. say today is 10/07/2024 and current valid subscription period is from 01/06/24 to 31/05/2025

then the membership_expiry_date will be the sub period end date and initial_membership date will be 01/07/2024 meaning the subscription will be for 11 months.